src/Store/VimStoreConnector.re

```diff
@@ -725,120 +774,28 @@ let start =
         isCompleting := true;
         let currentPos = ref(Vim.CommandLine.getPosition());
         while (currentPos^ > position) {
-          let _ = Vim.input(~cursors=[], "<bs>");
+          let _ = Vim.input("<bs>");
           currentPos := Vim.CommandLine.getPosition();
         };
 
         let completion = Path.trimTrailingSeparator(completion);
-        let latestCursors = ref([]);
+        let latestContext = ref(Vim.Context.default());
         String.iter(
           c => {
-            latestCursors := Vim.input(~cursors=[], String.make(1, c));
+            let (context, _eff) =
+              Vim.input(~context=latestContext^, String.make(1, c));
+            latestContext := context;
             ();
           },
           completion,
         );
```

Comment:
```suggestion
        String.fold(
          (c, context) => Vim.input(~context, String.make(1, c)) |> fst,
          Vim.Context.default(),
          completion,
        );
```

Reply:
I used `VimEx.inputString` since it handles UTF8 correctly, too